target/linux/ramips/dts/ArcherC2.dts

```diff
@@ -0,0 +1,168 @@
+/dts-v1/;
+
+/include/ "mt7620a.dtsi"
+
+/ {
+	compatible = "ralink,mt7620a-soc";
+	model = "TP-Link Archer C2";
+
+	chosen {
+		bootargs = "console=ttyS0,115200";
+	};
+
+
+	pinctrl {
+		state_default: pinctrl0 {
+			gpio {
+				ralink,group = "i2c", "uartf", "wled", "nd_sd", "ephy", "spi refclk";
+				ralink,function = "gpio";
+			};
+		};
+	};
+
+	gpio-leds { 
+		compatible = "gpio-leds";
+
+		lan {
+			label = "c2:green:lan";
+			gpios = <&gpio0 1 1>;
+		};
+		usb {
+			label = "c2:green:usb";
+			gpios = <&gpio0 11 1>;
+		};
+		wps {
+			label = "c2:green:wps";
+			gpios = <&gpio1 15 1>;
+		};
+		wan {
+			label = "c2:green:wan";
+			gpios = <&gpio2 0 1>;
+		};
+		wlan {
+			label = "c2:green:wlan";
+			gpios = <&gpio3 0 1>;
+		};
+	};
+
+	gpio-keys {
+		compatible = "gpio-keys";
+
+		reset_wps { 
+			label = "reset_wps"; 
+			gpios = <&gpio0 13 1>; 
+			linux,code = <0x198>; 
+		}; 
+
+		rfkill { 
+			label = "rfkill"; 
+			gpios = <&gpio0 2 1>; 
+			linux,code = <0xf7>; 
+		}; 
+	};
+
+	rtl8367rb {
+		compatible = "realtek,rtl8367b", "rtl8367b";
+		cpu_port = <6>;
+		realtek,extif1 = <1 0 1 1 1 1 1 1 2>;
+		mii-bus = <&mdio0>;
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	m25p80@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
+
+		partition@0 {
+			label = "u-boot";
+			reg = <0x0 0x20000>;
+			read-only;
+		};
+
+		partition@20000 {
+			label = "firmware";
+			reg = <0x20000 0x7a0000>;
+		};
+
+		partition@7c0000 {
```

Comment:
all partitions beginning from here should be read-only. I don't see a reason to change the contents of the partitions.